src/ext/property-panel/__tests__/data.test.ts

```diff
@@ -46,15 +53,59 @@ describe('data', () => {
       });
     });
 
+    describe('unsupportedFeature', () => {
+      let itemData: { qOtherTotalSpec: { qTotalMode: string } };
+      let args: {
+        properties: {
+          qHyperCubeDef: {
+            qIndentMode: boolean;
+          };
+        };
+      };
+      beforeEach(() => {
+        itemData = { qOtherTotalSpec: { qTotalMode: 'TOTAL_EXPR' } } as EngineAPI.IHyperCubeDimensionDef;
+        args = {
+          properties: {
+            qHyperCubeDef: {
+              qIndentMode: false,
+            },
+          },
+        } as Args;
+      });
+
+      afterAll(() => {
+        data = createData({
+          translator: { get: (str) => str },
+          anything: { sense: { isUnsupportedFeature: () => false } },
+        });
+      });
+
+      test('should not show totalMode and totalsLabel when feature is not supported', () => {
+        data = createData({
+          translator: { get: (str) => str },
+          anything: { sense: { isUnsupportedFeature: (f) => f === 'limitation' } },
+        });
+        expect(data.items.dimensions.items.totalMode.show).toBe(false);
+        expect(data.items.dimensions.items.totalsLabel.show(itemData, null, args)).toBe(false);
+      });
+      test('should show totalMode and totalsLabel when unsupported feature is not limitation', () => {
+        data = createData({
+          translator: { get: (str) => str },
+          anything: { sense: { isUnsupportedFeature: (f) => f === 'somethingElse' } },
+        });
+        expect(data.items.dimensions.items.totalMode.show).toBe(true);
+        expect(data.items.dimensions.items.totalsLabel.show(itemData, null, args)).toBe(true);
+      });
+
     describe('totalsLabel', () => {
       test('should show totalsLabel', () => {
         const itemData = { qOtherTotalSpec: { qTotalMode: 'TOTAL_EXPR' } } as EngineAPI.IHyperCubeDimensionDef;
         const args = {
           properties: {
             qHyperCubeDef: {
-              qIndentMode: false
-            }
-          }
+              qIndentMode: false,
+            },
+          },
```

Comment:
my VScode made some `Prettier` things to some of this code lines... pls skip these.